src/Management/src/Endpoint/ManagementWebHostBuilderExtensions.cs

```diff
@@ -313,6 +318,66 @@ public static IWebHostBuilder AddCloudFoundryActuator(this IWebHostBuilder hostB
     /// </param>
     public static IWebHostBuilder AddWavefrontMetrics(this IWebHostBuilder hostBuilder)
     {
-        return hostBuilder.ConfigureServices((_, collection) => collection.AddWavefrontMetrics());
+        return hostBuilder.ConfigureServices((context, collection) => collection.AddWavefrontMetrics());
+    }
+
+    internal static void GetManagementUrl(this IWebHostBuilder webHostBuilder, out int? httpPort, out int? httpsPort)
+    {
+        string portSetting = webHostBuilder.GetSetting(ManagementPortKey);
+        string sslSetting = webHostBuilder.GetSetting(ManagementSSLKey);
+
+        httpPort = httpsPort = null;
+
+        if (string.IsNullOrEmpty(portSetting))
```

Comment:
From reviewing this PR, I don't see a test or known scenario when `GetConfiguration()` should be used. I recall you mentioned there are scenarios where `GetSetting` on ln 326-327 doesn't return settings that have been specified, but I don't recall when that was - should we have a test or at least some commentary to indicate when that is expected to happen?

Reply:
So using the generic host (as all our old examples do) will need to build from appsettings because of the time at which this setting is needed. 
Although appsettings.json is always present in WebHost or WebApplicationBuilder usage, it is not available in a testing scenario and needs to be handled for all existing unit tests that use AddXActuator code. 
So about the unit test - its more of an integration test to test that there is an appsettings.json and it is formatted correctly and we would be basically testing framework code thats doing this:
`new ConfigurationBuilder().AddJsonFile("appsettings.json").Build();`  

I am open to suggestions to do this better though.

Reply:
Maybe adjusting the name `GetConfiguration` to something like `GetConfigurationFallback` or somehow indicate with the method name when it's expected to happen? If I look at this again in 6 months, I don't think I'll remember this detail from reading the code. 

If we absolutely must have this method, maybe it should match the default configuration expectations for WebHost? Like add Env vars and `appsettings.{Environment}.json` too?

Reply:
Done ...